src/activities/recent/recent_activities_dropdown.gd

```diff
@@ -0,0 +1,97 @@
+extends Container
+class_name RecentActivitiesDropdown
+
+static var SCENE: PackedScene = preload("res://src/activities/recent/RecentActivitiesDropdown.tscn")
+
+signal suggestion_accepted(suggestion: String)
+
+static var recent_cutoff: int = 10
+
+@onready var suggestions_container: VBoxContainer = $MarginContainer/SuggestionsContainer
+
+var recent_activities: Array[String] = []
+var suggestion_nodes: Array[Suggestion] = []
+
+func _ready() -> void:
+	_add_unique_activities_from_last_days(recent_cutoff)
+	suggestion_nodes = map_suggestions_to_node(recent_activities)
+	visible = is_suggestion_displayed()
+
+func _input(event: InputEvent) -> void:
+	if not visible:
+		return
+	if event.is_action_pressed("accept_suggestion"):
+		suggestion_accepted.emit(_find_selected())
+		get_viewport().set_input_as_handled()
+	elif event.is_action_pressed("next_suggestion"):
+		get_viewport().set_input_as_handled()
+		var suggestions: Array[Node] = suggestions_container.get_children()
+		var counter: int = 0
+		for suggestion: Suggestion in suggestions:
+			counter += 1
+			if suggestion.is_selected() and counter <= suggestions.size() - 1:
+				suggestion.selected = false
+				suggestions[counter].selected = true
+				return
+	elif event.is_action_pressed("previous_suggestion"):
+		get_viewport().set_input_as_handled()
+		var suggestions: Array[Node] = suggestions_container.get_children()
+		var counter: int = 0
+		for suggestion: Suggestion in suggestions:
```

Comment:
instead of using foreach and counter variable, use for loop?

Reply:
fair point. guess I'm not python enough to think of this syntax